Tws scramble

tedsharpe opened the pull request.

mwalker174 commented on wdl/GatherSampleEvidenceMetrics.wdl:
```diff
@@ -141,6 +144,19 @@ workflow GatherSampleEvidenceMetrics {
+        baseline_vcf = select_first([baseline_scramble_vcf]),
+        samples = [sample],
+        prefix = "scramble_" + sample,
+        types = "DEL,INS",
```
```suggestion
        types = "INS",
```

mwalker174 commented on wdl/AnnotateILFeatures.wdl:
```diff
@@ -225,7 +225,6 @@ workflow AnnoILFeatures {
     call MiniTasks.ConcatVcfs as ConcatVcfsIL{
         input:
             vcfs=Bcf2VcfIL.vcf,
-            merge_sort=true,
```
Looks like this is reverting a recent change?

tedsharpe replied:
The task ConcatVcfs in TasksMakeCohortVcf.wdl doesn't have that argument in master.  Is there some other branch that isn't checked in yet that I ought to be basing this on?

mwalker174 commented on wdl/Duphold.wdl:
```diff
@@ -75,7 +75,6 @@ workflow Duphold {
   call MiniTasks.ConcatVcfs as ConcatVcfs{
     input:
       vcfs=Bcf2Vcf.vcf,
-      merge_sort=true,
```
Another reversion?

mwalker174 commented on wdl/GatherSampleEvidence.wdl:
```diff
@@ -319,6 +339,9 @@ workflow GatherSampleEvidence {
     Int? melt_read_length = MELT.read_length_out
     Float? melt_insert_size = MELT.insert_size_out
 
+    File? scramble_vcf = Scramble.insertions_vcf
+    File? scramble_index = Scramble.insertions_index
```
Unless I'm missing something, it doesn't look like the deletions vcf was ever hooked up here? Does this mean your batch run with deletions enabled wasn't correct?

tedsharpe replied:
Damn.  Yeah, it probably does.

mwalker174 replied:
Ok, let's not bother with running that again. The runtimes are too high and variable with deletion calling.

tedsharpe commented on wdl/Scramble.wdl:
```diff
@@ -0,0 +1,188 @@
+## https://github.com/GeneDx/scramble
+##
+## Requirements/expectations :
+## - Human whole-genome pair-end sequencing data in mapped BAM format
```
Don't think so, but I haven't tried.

mwalker174 replied:
I think it can be: https://github.com/GeneDx/scramble/blob/14806b4569d20e85835ace7f0e32419085d9bae3/cluster_identifier/src/cluster_identifier.c#L77

tedsharpe replied:
Hah!  Pretty obscure, but looks like you're right.

tedsharpe commented on wdl/Scramble.wdl:
```diff
@@ -0,0 +1,188 @@
+  }
+    
+  parameter_meta {
+    bam_or_cram_file: "A .bam or .cram file to search for SVs. crams are preferable because they localise faster and use less disk."
```
Don't think that works.  See above.

tedsharpe commented on wdl/Scramble.wdl:
```diff
@@ -0,0 +1,188 @@
+    meiRef=$scrambleDir/cluster_analysis/resources/MEI_consensus_seqs.fa
+
+    # create a blast db from the reference
+    cat ~{reference_fasta} | makeblastdb -in - -parse_seqids -title ref -dbtype nucl -out ref
```
Seconds.  Not necessary.

tedsharpe commented on wdl/Scramble.wdl:
```diff
@@ -0,0 +1,188 @@
+    mem_gb: 16,
+    disk_gb: 250,
+    boot_disk_gb: 10,
+    preemptible_tries: 1,
```
With deletions the times vary quite a bit:
count hours
      3 13
     24 14
     21 15
     12 16
      8 17
      3 18
      1 19
      2 21
But my impression is that in MEI-only mode, the times are small and pretty consistent, so I'm adopting your proposal.

tedsharpe commented:
I think I've addressed all your comments.
I'm running one sample with deletions on just to make sure that I haven't messed that up in Scramble.wdl.  If that's successful, I think we might be ready to merge pending additional comments.
